tests/encfs/test.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import base64
+import binascii
+import json
+import subprocess
+import tempfile
+import requests
+import uuid
+import os
+import unittest
+import sys
+
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+from encfs import deploy_encfs
+from skr.key import generate_key, deploy_key
+
+from c_aci_testing.aci_is_live import aci_is_live
+from c_aci_testing.aci_param_set import aci_param_set
+from c_aci_testing.images_build import images_build
+from c_aci_testing.images_push import images_push
+from c_aci_testing.policies_gen import policies_gen
+from c_aci_testing.target_run import target_run_ctx
+from c_aci_testing.aci_get_ips import aci_get_ips
+
+class EncFSTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        target_dir = os.path.realpath(os.path.dirname(__file__))
+        id = os.getenv("ID", str(uuid.uuid4()))
+        tag = os.getenv("TAG") or id
+
+        attestation_endpoint = os.environ["ATTESTATION_ENDPOINT"]
+        hsm_endpoint = os.environ["HSM_ENDPOINT"]
+        storage_account_name = os.environ["STORAGE_ACCOUNT_NAME"]
+        storage_container_name = os.environ["STORAGE_CONTAINER_NAME"]
+        key_id = f"{id}-key"
+        cls.test_file_content = "Hello, World!"
+        mount_point = "/mnt/remote"
+        cls.blobs = [
+            (f"{id}-blob1", "page"),
+            (f"{id}-blob2", "block"),
+        ]
+
+        azure_args = {
+            "subscription": os.getenv("SUBSCRIPTION"),
+            "resource_group": os.getenv("RESOURCE_GROUP")
+        }
+
+        image_args = {
+            "target": target_dir,
+            "registry": os.environ["REGISTRY"],
+            "repository": os.getenv("REPOSITORY"),
```

Comment:
nit: env vars could be validated before using them. if this isn't a risk, I think it's fine though.

Reply:
If you mean validated that they exist, I've used os.environ for registry for that exact reason so we get a keyerror if it's not set. For repository the testing package allows None which is why I use getenv() instead